workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/ComposeRendering.kt

```diff
@@ -0,0 +1,68 @@
+package com.squareup.workflow1.ui.compose
+
+import androidx.compose.runtime.Composable
+import com.squareup.workflow1.ui.AndroidViewRendering
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewFactory
+import com.squareup.workflow1.ui.ViewRegistry
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import kotlin.reflect.KClass
+
+/**
+ * Interface implemented by a rendering class to allow it to drive a composable UI via an
+ * appropriate [ComposeViewFactory] implementation, by simply overriding the [Content] method.
+ * This is the compose analog to [AndroidViewRendering].
+ *
+ * Note that unlike most workflow view functions, [Content] does not take the rendering as a
+ * parameter. Instead, the rendering is the receiver, i.e. the current value of `this`.
+ *
+ * Example:
+ *
+ * ```
+ * @OptIn(WorkflowUiExperimentalApi::class)
+ * data class HelloView(
+ *   val message: String,
+ *   val onClick: () -> Unit
+ * ) : ComposeRendering {
+ *
+ *   @Composable override fun Content(viewEnvironment: ViewEnvironment) {
+ *     Button(onClick) {
+ *       Text(message)
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * This is the simplest way to bridge the gap between your workflows and the UI, but using it
+ * requires your workflows code to reside in Android modules, instead of pure Kotlin. If this is a
+ * problem, or you need more flexibility for any other reason, you can use [ViewRegistry] to bind
+ * your renderings to [ComposeViewFactory] implementations at runtime.
+ */
+@WorkflowUiExperimentalApi
+public interface ComposeRendering : AndroidViewRendering<Nothing> {
+
+  /** Don't override this, override [Content] instead. */
+  override val viewFactory: ViewFactory<Nothing> get() = Companion
```

Comment:
The ViewRegistry overhaul I've been trying to get back to since November(!) allows us to provide bindings for factories of arbitrary types, e.g. DialogFactory in addition to ViewFactory. When I finally get back to that (this quarter or bust), we can also make it vend a ComposableFactory and get rid of this `ViewFactory<Nothing>` wart.